docs/site/Req-res-cycle.md

```diff
@@ -0,0 +1,278 @@
+`@put` etc., create endpoints on the app for the corresponding HTTP verbs. The
+behavior of these endpoints are entirely dependent on the implementation of the
+controller method.
+
+{% include tip.html content="Apart from controller files in the `controllers` directory,
+controllers may be added to the app by [components](https://loopback.io/doc/en/lb4/Components.html)." %}
+
+In the request-response cycle section we will see how implemenation details
+determine the course of a request to these endpoints - they may or may not actually
+interact with a model.
+
+#### Non-controller endpoints
+
+The following APIs can create non-controller endpoints.
+
+##### 1. app.static(path, rootDir, options)
+
+This method maps to the underlying Express' [static](https://expressjs.com/en/4x/api.html#express.static)
+method. It creates dynamic endpoints for files in the `rootDir` to be served
+at `path`.
+
+More details at [RestApplication.static()](https://loopback.io/doc/en/lb4/apidocs.rest.restapplication.static.html)
+doc.
+
+##### 2. app.redirect(fromPath, toPathOrUrl, statusCode)
+
+This method redirects requests to `fromPath` to `toPathOrUrl` with the status
+code `statusCode`.
+
+More details at [RestApplication.redirect()](https://loopback.io/doc/en/lb4/apidocs.rest.restapplication.redirect.html)
+doc.
+
+##### 3. app.mountExpressRouter(basePath, router, spec)
+
+This method mounts an Express [router](https://expressjs.com/en/4x/api.html#router)
+in the LoopBack app. You can use it for adding custom endpoints in your app using
+the Express middleware format.
+
+More details at [RestApplication.mountExpressRouter()
+](https://loopback.io/doc/en/lb4/apidocs.rest.restapplication.mountexpressrouter.html) doc.
+
+##### 4. non-controller endpoints added by components
+
+[Components](https://loopback.io/doc/en/lb4/Components.html) are the extension
+units of LoopBack. Since they can access the LoopBack app instance, the request
+and response instances using [dependency injection](https://loopback.io/doc/en/lb4/Dependency-injection.html),
+they are capable of adding new endpoints and determining the result of a request.
+
+### The request-response cycle
+
+The request-response cycle involves many components, especially if the request
+is to a controller-endpoint which interacts with the database.
+
+![Components of LoopBack 4 request-response cycle](./imgs/req-res-high-level.png)
+
+The request handling process starts with the app's [sequence](https://loopback.io/doc/en/lb4/Sequence.html);
+it is the gatekeeper of all requests to the app. Every request, whether to
+controller endpoints or non-controller endpoints, must pass through the sequence.
+
+The sequence identifies the responsible handler for the requested endpoint and
+passes on the request to the handler. The handler then take care of sending
+the response back to the client.
+
+#### The sequence
+
+The sequence is a simple class with five injected helper methods. These five
```

Comment:
`with five injected helper methods` -> `with five injected helper methods by default`?

Reply:
Made it `five injected helper methods in the constructor.`